### DISCUSSION TITLE
Improved gov module

### DISCUSSION THREAD

**Discussion by kimurayu45z** [cosmos/community#96] [Cosmos SDK]
https://x.com/zmanian/status/1841120412320874755

In this perspective, giving voting powers for governance proposal to validators that are derived from delegations should be eliminated from gov module.
Only the voting powers for consensus process should take delegation in consideration.

I think it is not too difficult to develop such a custom gov module.

**Comment by julienrbrt** [comment]
Hey! This is actually supported in the next version of gov that is 0.52 compatible:

- https://github.com/cosmos/cosmos-sdk/blob/v0.52.0-beta.1/x/gov/keeper/config.go#L33-L35
- https://github.com/cosmos/cosmos-sdk/blob/v0.52.0-beta.1/x/gov/keeper/tally.go#L235-L325

The default tally function, stays as in, and keeps validators voting power counting towards the vote outcome.
However, if you define another `CalculateVoteResultsAndVotingPowerFn` to avoid counting derived validators votes, that is as well possible. Given that you have access to the proposal, you could implement the logic you propose (count them only for upgrade proposal or parameter upgrade proposals for instance)
